src/xenia/app/xenia_main.cc

```diff
@@ -278,27 +278,22 @@ int xenia_main(const std::vector<std::string>& args) {
   }
 
   if (cvars::mount_cache) {
-    auto cache0_device =
-        std::make_unique<xe::vfs::HostPathDevice>("\\CACHE0", "cache0", false);
-    if (!cache0_device->Initialize()) {
-      XELOGE("Unable to scan cache0 path");
-    } else {
-      if (!emulator->file_system()->RegisterDevice(std::move(cache0_device))) {
-        XELOGE("Unable to register cache0 path");
-      } else {
-        emulator->file_system()->RegisterSymbolicLink("cache0:", "\\CACHE0");
-      }
-    }
+    constexpr std::pair<std::string_view, std::string_view> cache_mounts[] = {
+        {"\\CACHE", "cache"}, {"\\CACHE0", "cache0"}, {"\\CACHE1", "cache1"}};
 
-    auto cache1_device =
-        std::make_unique<xe::vfs::HostPathDevice>("\\CACHE1", "cache1", false);
-    if (!cache1_device->Initialize()) {
-      XELOGE("Unable to scan cache1 path");
-    } else {
-      if (!emulator->file_system()->RegisterDevice(std::move(cache1_device))) {
-        XELOGE("Unable to register cache1 path");
+    for (auto const cache_mount : cache_mounts) {
+      auto cache_device = std::make_unique<xe::vfs::HostPathDevice>(
```

Comment:
A special cache directory (`cache_root`) has recently been added — please use it instead so it can be cleared from the system settings on Android in the future (something like `{cache_root}/guest_mounts/{title ID}/…`, though a more nice subdirectory name may be chosen, it's up for discussion).

Reply:
Making cache per-titleID is a good idea, on actual 360 it clears cache between different titles IIRC (and maybe media IDs too), and some games can do weird things if cache has data it doesn't expect (eg. Halo 3 delta has issues loading if you use cache from even a different build-type (debug/release/etc) than the one you're using, titleid wouldn't help here though, but maybe also using mediaid would?)

I guess cache-mounting code would need to be moved from xenia_main() for this though, that'd probably be helpful anyway, ran into problems with xenia only creating the cache folders on startup instead of when a game is actually loaded too many times now...

Reply:
If it's that sensitive we may want to generate a hash of the executable's code for uniqueness then.

Reply:
Yeah that'd probably be best, just checked the halo 3 delta XEXs and none of them seem to use media IDs at all unfortunately. Maybe there's a hash in the XEX headers we could use instead of needing to generate it ourselves.

Reply:
> Making cache per-titleID is a good idea, on actual 360 it clears cache between different titles IIRC

If it is cleared everytime you run different app, then maybe we should just clear it while xenia is opened? 🤔 It's only a game cache tbh and it shouldn't (?) have performance impact.

Reply:
If we make an implementation where the cache only lives for as long as the game is running, maybe a RAM-based virtual file system could also be a good option (along with a true directory on the disk).